main_test.go

```diff
@@ -211,11 +211,11 @@ func registeredTestWslInstances() ([]wsl.Distro, error) {
 func defaultDistro() (string, error) {
 	out, err := exec.Command("powershell.exe", "-Command", "$env:WSL_UTF8=1; wsl.exe --list --verbose").CombinedOutput()
 	if err != nil {
-		if !errors.Is(err, &exec.ExitError{}) {
+		if !errors.As(err, new(*exec.ExitError)) {
```

Comment:
As I told you earlier today, the `new` is not really something that is encouraged within the Go community. Here, it’s worse, you have a `new(<pointer>something)` as exec returns a pointer to the ExitError (which I think, is due to the `*os.ProcessState` it’s olding, to not give the bad idea that you can safely mess with your own copy when it’s holding a reference to the underlying process, but that’s really my guess).

So, you are really correct that we can’t use `Is` because of the internal fields that don’t match. I would rewrite that `As` condition as follow to be more in line with the traditional practice of `As`, even if we don’t reuse the value:
`if e := (&exec.ExitError{}); errors.As(err, &e) {`

This is a suggestion, do what you want with it :)

Reply:
It looks cleaner, I applied this change.